invenio_communities/communities/services/service.py

```diff
@@ -127,7 +110,9 @@ def search_user_communities(
         # Prepare and execute the search
         params = params or {}
 
-        current_user_filter = CommunityMembers().query_filter(identity)
+        current_user_filter = CommunityMembers().query_filter(identity) & dsl.Q(
+            "term", **{"deletion_status": CommunityDeletionStatusEnum.PUBLISHED.value}
+        )
```

Comment:
this will certainly hide deleted records from the search results for the user's communities (without it, they'd still pop up)

maybe this should be done via generator query filters somehow?

Reply:
PIng @kpsherva